caffe2/binaries/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ if (USE_CUDA)
   if (BUILD_TEST)
     # Core overhead benchmark
     caffe2_binary_target("core_overhead_benchmark.cc")
-    target_link_libraries(core_overhead_benchmark benchmark)
+    target_link_libraries(core_overhead_benchmark benchmark ${CUDA_curand_LIBRARY})
```

Comment:
Thanks! Actually, I think it might need other cuda libraries as well so it would be probably easier to link against caffe2_gpu?

Reply:
Yeah, agree. I was wondering why the OSS CI didn't catch this building issue. I caught it because the conda build failed. ;-)

Reply:
Actually I am having now problems with building caffe2 (ed1b7d9b) and it fails nearly at the end:
```
[ 99%] Linking CXX executable ../../bin/core_overhead_benchmark
CMakeFiles/core_overhead_benchmark.dir/core_overhead_benchmark.cc.o: In function `BM_cudaStreamWaitEventThenStreamSynchronize(benchmark::State&)':
core_overhead_benchmark.cc:(.text+0x2e2): undefined reference to `cudaStreamCreate'
core_overhead_benchmark.cc:(.text+0x32f): undefined reference to `cudaEventCreateWithFlags'
core_overhead_benchmark.cc:(.text+0x36d): undefined reference to `cudaEventRecord'
core_overhead_benchmark.cc:(.text+0x3ad): undefined reference to `cudaStreamWaitEvent'
...
```

All the missing functions:
```
cudaEventCreateWithFlags
cudaEventRecord
cudaGetDevice
cudaGetErrorString
cudaGetLastError
cudaSetDevice
cudaStreamCreate
cudaStreamCreateWithFlags
cudaStreamDestroy
cudaStreamSynchronize
cudaStreamWaitEvent
```
can be found in libcudart_static.a but I don't know how to specify it in this CMakeLists.txt.
I'd appreciate your hits :)

Best regards
Andrzej

Reply:
I've found the solution now.  In FindCUDA.cmake `CUDA_LIBRARIES` is described as "Cuda RT library" so I've just appended it and now everything builds OK.  You might use the attached patch to fix it.

Best regards
Andrzej
[diff.txt](https://github.com/caffe2/caffe2/files/1497548/diff.txt)